pkg/cmd/pipeline/translate.go

```diff
@@ -189,6 +195,43 @@ func UpdateEnvs(ctx context.Context, name, namespace string, envs []string, c ku
 	return nil
 }
 
+// AddPhaseDuration adds a new phase to the configmap with the duration in seconds
+func AddPhaseDuration(ctx context.Context, name, namespace, phase string, duration time.Duration, c kubernetes.Interface) error {
+	cmap, err := configmaps.Get(ctx, TranslatePipelineName(name), namespace, c)
+	if err != nil {
+		return err
+	}
+	val, ok := cmap.Data[phasesField]
```

Comment:
Given that right now we are only adding the phases field in order to track events and they are calculated and created in K8s in the CLI execution, there is no scenario where we might lose the information. 

If we wanted to store the information to show it on the UI, there is one scenario that we have to take a look (deploying with a previous CLI version)

Reply:
Given that right now we are only adding the phases field in order to track events and they are calculated and created in K8s in the CLI execution, there is no scenario where we might lose the information. 

If we wanted to store the information to show it on the UI, there is one scenario that we have to take a look (deploying with a previous CLI version)

Reply:
Would there be an scenario in which a deploy within another deploy would use the same configmap, overriding the `phase` field for the main one?

Reply:
Do you mean if they have the same name? It'll create two events with the same information besides the duration. What it will happen is:

1. Main deploy starts the timer:
2. The deploy from inside starts the timer
3. The deploy from inside updates the configmap
4. The deploy from inside sends the event with its duration
5. The main deploy updates the configmap
6. The main deploy sends the event with its duration

I think it's the way it must behave but open to discuss it

Reply:
In the moment in which the main deploy add a phase before the "deploy inside" adds its own, the inside one would overwrite the `phase` field of the main one. But not sure if the scenario is possible or not